test/test_watch_notice.rb

```diff
@@ -0,0 +1,16 @@
+require 'test_helper'
+
+class TestWatchNotice < MiniTest::Test
+  def test_recurse_over_arrays
+    json = JSON.parse(open_test_file('node_notice.json').read)
+    notice = Kubeclient::Common::WatchNotice.new(json)
+    assert_kind_of(Array, notice.object.status.addresses)
+    notice.object.status.addresses.each do |address|
+      assert_kind_of(OpenStruct, address)
```

Comment:
s/OpenStruct/RecursiveOpenStruct/ 
Will be helpful to do the same in the commit message